File: core/src/no/elg/infiniteBootleg/Ticking.java
package no.elg.infiniteBootleg;

import no.elg.infiniteBootleg.util.Ticker;

/**
 * @author Elg
 */
public interface Ticking {

    /**
     * Tick an object. This will be called in sync with the current {@link Ticker}
     */
    void tick();

    /**
     * Update rarely (for expensive methods) this should be called every {@link Ticker#getTickRareRate()} ticks
     */
    default void tickRare() {}
}

File: core/src/main/kotlin/no/elg/infiniteBootleg/api/Ticking.kt
package no.elg.infiniteBootleg.api

/**
 * @author Elg
 */
interface Ticking {
  /**
   * Tick an object.
   *
   * You might want to use synchronization either on the whole method or parts of the method to
   * ensure correctness
   */
  fun tick()

  /**
   * Update rarely (for expensive methods) this should be called every [Ticker.getTickRareRate] ticks.
   *
   *
   * You might want to use synchronization either on the whole method or parts of the method to
   * ensure correctness
   */
  fun tickRare() {}
}
